File: public/examples/integrations/toolkits/google_calendar/find_time_slots_when_everyone_is_free_example_call_tool.js
import { Arcade } from "@arcadeai/arcadejs";

const client = new Arcade(); // Automatically finds the `ARCADE_API_KEY` env variable

const USER_ID = "{arcade_user_id}";
const TOOL_NAME = "GoogleCalendar.FindTimeSlotsWhenEveryoneIsFree";

// Start the authorization process
const authResponse = await client.tools.authorize({tool_name: TOOL_NAME});

if (authResponse.status !== "completed") {
  console.log(`Click this link to authorize: ${authResponse.url}`);
}

// Wait for the authorization to complete
await client.auth.waitForCompletion(authResponse);

const toolInput = {
  "email_addresses": [
    "alice@example.com",
    "bob@example.com"
  ],
  "start_date": "2023-10-01",
  "end_date": "2023-10-07",
  "start_time_boundary": "09:00",
  "end_time_boundary": "17:00"
};

const response = await client.tools.execute({
  tool_name: TOOL_NAME,
  input: toolInput,
  user_id: USER_ID,
});

console.log(JSON.stringify(response.output.value, null, 2));

File: public/examples/integrations/toolkits/google_calendar/find_time_slots_when_everyone_is_free_example_call_tool.py
import json
from arcadepy import Arcade

client = Arcade()  # Automatically finds the `ARCADE_API_KEY` env variable

USER_ID = "{arcade_user_id}"
TOOL_NAME = "GoogleCalendar.FindTimeSlotsWhenEveryoneIsFree"

auth_response = client.tools.authorize(
    tool_name=TOOL_NAME,
    user_id=USER_ID
)

if auth_response.status != "completed":
    print(f"Click this link to authorize: {auth_response.url}")

# Wait for the authorization to complete
client.auth.wait_for_completion(auth_response)

tool_input = {
    'email_addresses': ['alice@example.com', 'bob@example.com', 'carol@example.com'],
    'start_date': '2025-09-15',
    'end_date': '2025-09-19',
    'start_time_boundary': '09:00',
    'end_time_boundary': '17:00'
}

response = client.tools.execute(
    tool_name=TOOL_NAME,
    input=tool_input,
    user_id=USER_ID,
)
print(json.dumps(response.output.value, indent=2))
